src/tbb/thread_data.h

```diff
@@ -204,6 +213,74 @@ class thread_data : public ::rml::job
     // TODO: consider using common default context because it is used only to simplify
     // cancellation check.
     d1::task_group_context my_default_context;
+#ifdef __TBB_STATISTICS
+    enum class Statistics : std::size_t {
+        EXEC_TASK = 0,
+        GET_TASK,
+        TASK_RES_NOT_NULLPTR,
+        STEAL_TASK,
+        TRY_STEAL_TASK,
+        VICTIM_POOL_EMPTY,
+        STEAL_RES_NOT_NULLPTR,
+        SIZE,
+    };
+
+    void add_statistics(Statistics stat_type) {
+        ++statistics[static_cast<std::size_t>(stat_type)];
+    }
+private:
+    using ArrayStat = std::array<std::uint32_t, static_cast<std::size_t>(thread_data::Statistics::SIZE)>;
+    ArrayStat statistics {}; // main thread statistics object
+
+    class CollectStatistics : no_copy {
+    public:
+        CollectStatistics() = default;
+
+        CollectStatistics(CollectStatistics&&) = delete;
+
+        CollectStatistics& operator=(CollectStatistics&&) = delete;
```

Comment:
no_copy and ` CollectStatistics(CollectStatistics&&) = delete;`  seems as duplication. Moreover, `std::mutex` is not copiable.